ols/app/models/models.py

```diff
@@ -103,7 +103,7 @@ class FeedbackRequest(BaseModel):
     """
 
     conversation_id: str
-    feedback_object: dict  # TODO: proper object (OLS-77)
+    feedback_object: dict  # TODO: proper object will be defined by OLS-138
```

Comment:
[OLS-138](https://issues.redhat.com/browse/OLS-138) is concluded, the information we want to collect:
* sentiment(thumbs up/down)
* a freeform text string
* the user query/question that was submitted
* the llm response to the question
* the conversation id

@thoraxe sound right?

(we want the user query+response separately despite having the conversation id(with which we could go look up the entire conversation via our transcript recording feature) because we want to see the exact part of the conversation they are providing feedback on, in the case of a conversation that spans multiple follow up queries.  It's also possible some customers will enable feedback reporting, but will disallow the general collection of transcripts.  In that case, this will allow us to still have at least some context for the feedback)

Reply:
If I get it right, those are basically two different feedbacks. One is +1/-1 to a response and the other is generic. So maybe accept two models by endpoint (to avoid allowing posting "nothing")?
```python
# thumbs feedback
conversation_id: str
user_question: str
llm_response: str
sentiment: int

# generic feedback
user_feedback: str
```
Or maybe better to split the endpoint into two.
WDYT?

Reply:
> Everything is optional because the freeform feedback is not bound to any conversation right? 

Feedback is only given in reaction to an exchange in a conversation.  Even the "Freeform" text is freeform text about the interaction.

they aren't two different kinds of feedback object.

```
conversationid: required str
user_question: required str
llm_response: required str
sentiment: required int
freeform_text: optional str
```

Reply:
Hmm, ok. So the user can either provide thumbs or text feedback for response. The `conversation_id`, `user_question` and `llm_response` are mandatory. And `sentiment` or `freeform_text` will be provided (are optional from the model point of view). Correct?

Reply:
> Hmm, ok. So the user can either provide thumbs or text feedback for response. The conversation_id, user_question and llm_response are mandatory. And sentiment or freeform_text will be provided (are optional from the model point of view). Correct?

well the way i wrote it above, thumbs up/down would be mandatory.  But thinking about it some more, i'm fine with it being optional.  (but if they don't specify sentiment and they don't specify freeform, that's an invalid feedback submission.  We need at least one of them)